operation/fhir-operation-everything/src/main/java/com/ibm/fhir/operation/everything/EverythingOperation.java

```diff
@@ -0,0 +1,309 @@
+        queryParametersWithoutDates.remove(DATE_QUERY_PARAMETER);
+        
+        List<String> resourceTypesOverride = getOverridenIncludedResourceTypes(parameters);
+        List<String> resourceTypes = resourceTypesOverride.isEmpty() ? defaultResourceTypes : resourceTypesOverride;
+        
+        for (String compartmentType : resourceTypes) {
+            MultivaluedMap<String, String> searchParameters = queryParameters;
+            if (!SUPPORT_CLINICAL_DATE_QUERY.contains(compartmentType)) {
+                searchParameters = queryParametersWithoutDates;
+            }
+            Bundle results = null;
+            try {
+                results = resourceHelper.doSearch(compartmentType, PATIENT, logicalId, searchParameters, null, null, null);
+            } catch (Exception e) {
+                LOG.warning("Error retrieving $everything resources of type " + compartmentType + " for patient " + logicalId);
+                continue;
+            }
+            allEntries.addAll(results.getEntry());
+        }
+        
+        Bundle.Builder bundleBuilder = Bundle.builder()
+                .type(BundleType.SEARCHSET)
+                .id(UUID.randomUUID().toString())
+                .entry(allEntries)
+                .total(UnsignedInt.of(allEntries.size()));        
+        
+        Parameters outputParameters;
+        try {
+            outputParameters = FHIROperationUtil.getOutputParameters(bundleBuilder.build());
+        } catch (Exception e) {
+            FHIROperationException exceptionWithIssue = buildExceptionWithIssue("An unexpected error occurred while creating the operation output parameters for the resulting Bundle.", IssueType.EXCEPTION);
+            LOG.throwing(this.getClass().getName(), "doInvoke", exceptionWithIssue);
+            throw exceptionWithIssue;
+        }
+        LOG.exiting(this.getClass().getName(), "doInvoke", outputParameters);
+        return outputParameters;
+    }
+
+    /**
+     * Parse the parameters and turn them into a {@link MultivaluedMap} to pass to the search service
+     * 
+     * @param parameters the operation parameters
+     * @return the {@link MultivaluedMap} for the search service built from the parameters 
+     */
+    protected MultivaluedMap<String, String> parseQueryParameters(Parameters parameters) {
+        MultivaluedMap<String, String> queryParameters = new MultivaluedHashMap<String, String>();
+        Parameter countParameter = getParameter(parameters, SearchConstants.COUNT);
+        if (countParameter == null) {
+            queryParameters.add(SearchConstants.COUNT, SearchConstants.MAX_PAGE_SIZE + "");
+        } else {
+            queryParameters.add(SearchConstants.COUNT, countParameter.getValue().as(FHIR_INTEGER).getValue() + "");
+        }
+        // We use gt/lt here in an effort to be more liberal in terms of what we return
```

Comment:
The _count parameter is problematic in that you can't take advantage of the persistence layer's paging since you're making a separate call per resource type. If you're not going to implement paging at this layer, I would make that clear in the documentation, and just set _count to MAX_PAGE_SIZE on all your calls to FHIRRestHelper.doSearch().

Reply:
Correct paging is not supported but they can provide a _count, I thought this matched the spec.

Reply:
I think this works ok. If we're not supporting paging and we interpret _count as the max to be returned for the entire bundle, then we should never get more than _count resources per resource type.